### ISSUE TITLE
Hubbard model: mutual information

### ISSUE THREAD

**Issue by xdvriend** [GQCG/GQCP#876]
For the constrained entanglement research we need a new feature: the ability to calculate the mutual information between sites of the Hubbard model.

This feature has not yet been implemented in GQCP.

**Comment by xdvriend** [comment]
@dariatols I believe we can close this issue right? Since its intent has been covered in GQCP/GQCD?

**Comment by dariavh** [comment]
Indeed! However, we should check the values with some reference data first.
